File: test-suite-kotlin/src/test/kotlin/io/micronaut/rxjava3/docs/HelloController.kt
package io.micronaut.rxjava3.docs

import io.micronaut.context.annotation.Requires
import io.micronaut.http.MediaType
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get
import io.micronaut.http.sse.Event
import io.reactivex.rxjava3.core.Flowable

@Requires(property = "spec.name", value = "Rx3HttpClientTest")
@Controller("/hello")
class HelloController {

    @Get(produces = [MediaType.TEXT_PLAIN])
    fun hello(): String = "Hello World"

    @Get("/stream", produces = [MediaType.APPLICATION_JSON_STREAM])
    fun stream(): Flowable<Message> = Flowable.just(Message("Hello"), Message("World"))

    @Get("/events", produces = [MediaType.TEXT_EVENT_STREAM])
    fun events(): Flowable<Event<String>> = Flowable.just(Event.of("Hello"), Event.of("World"))
}

File: test-suite-python/src/test/python/micronaut/rxjava3/docs/HelloController.py
from micronaut.context.annotation import Requires
from micronaut.http import MediaType
from micronaut.http.annotation import Controller, Get
from micronaut.http.sse import Event
from io.reactivex.rxjava3.core import Flowable

from .Message import Message


@Requires(property="spec.name", value="Rx3HttpClientTest")
@Controller("/hello")
class HelloController:

    @Get(produces=MediaType.TEXT_PLAIN)
    def hello(self) -> str:
        return "Hello World"

    @Get(value="/stream", produces=MediaType.APPLICATION_JSON_STREAM)
    def stream(self) -> Flowable[Message]:
        return Flowable.just(Message("Hello"), Message("World"))

    @Get(value="/events", produces=MediaType.TEXT_EVENT_STREAM)
    def events(self) -> Flowable[Event[str]]:
        return Flowable.just(Event.of("Hello"), Event.of("World"))
